CRM/Core/Payment/OmnipayMultiProcessor.php

```diff
@@ -360,6 +360,12 @@ private function getCreditCardObjectParams($params) {
     if (is_numeric($cardFields['billingState'])) {
       $cardFields['billingState'] = CRM_Core_PseudoConstant::stateProvince($cardFields['billingState']);
     }
+    // Some processors (eg. sagepay) require that fields that are empty are not set
+    foreach ($cardFields as $name => $value) {
+      if (empty($value)) {
+        unset($cardFields[$name]);
+      }
+    }
```

Comment:
@eileenmcnaughton Is this something that's acceptable for all payment processors?